### PR TITLE
[qa] Avoid leaving test files after tests

### LINKED ISSUE #59: [qa] Avoid leaving test files after tests
**Reserved for GCI**

After running tests a test csv file is left on the filesystem.

- configure the `settings.py` test environment so that media files are uploaded to `./tests/media/`
- add `./tests/media/*` to `.gitignore`
- make sure `./tests/media/` is cleaned up during `tearDown`

### PR DESCRIPTION
**Pull request by cking100**
Fixes #59

### CODE REVIEW

**Comment by cking100** [comment]
@nemesisdesign By default, the csv file is genearated inside parent directory itself. i tried changing that using `MEDIA_ROOT` and `MEDIA_URL` but that is not working. I think this is how its done in other repos as well so what may be the cause?

File: tests/urls.py [hunk 1]
```diff
@@ -1,10 +1,18 @@
...
 openwisp_admin()
 admin.autodiscover()
 
 urlpatterns = [
     url(r'^', include('django_ipam.urls', namespace='ipam')),
     url(r'^admin/', admin.site.urls),
 ]
+
+urlpatterns += staticfiles_urlpatterns()
+urlpatterns += static(settings.MEDIA_URL, document_root=settings.MEDIA_ROOT)
```
Comment (nemesifier) [inline tests/urls.py]: not sure this addition is needed, but let's keep it on hold until we resolve the doubts

**Review by nemesifier — CHANGES_REQUESTED** [review]
@cking100 can you point out where the file is generated?

Right now I don't remember the answers to this question:

- Is it a file generated by the system or is it uploaded by the user?
- if it's uploaded by the user, do we really need to store it? If it's needed for importing data, we probably don't have to store it at all and we can get rid of it as soon as we have read it

Help me determine the answer to these questions and we'll then know what the right solution is.

PS: fix the flake8 warnings please

**Comment by cking100** [comment]
@nemesisdesign A `data_123...` file is created in the main directory itself under `~/django-ipam/` and this file is used here https://github.com/openwisp/django-ipam/blob/a74ab5d115c2421007e3491a50394f6dac93af06/django_ipam/management/commands/__init__.py#L46 So, i think it is necessary to store it and i'm not sure but i think it is created by the system every time after running tests

**Review by nemesifier — COMMENTED** [review]
@cking100 great, try adding a `tearDown` method to this class:
https://github.com/openwisp/django-ipam/blob/master/django_ipam/tests/base/test_commands.py

Make the method cleanup a pattern that match that path/filename and nothing else.

**Review by nemesifier — APPROVED** [review]
:+1:

File: django_ipam/tests/base/test_commands.py [hunk 2]
```diff
@@ -30,3 +30,10 @@ def test_import_subnet_command(self):
             call_command('import_subnet', file='invalid.pdf')
         with self.assertRaises(CommandError):
             call_command('import_subnet', file='invalid_path.csv')
+
+    def tearDownClass():
+        import glob
+        import os
```
Comment (nemesifier) [inline django_ipam/tests/base/test_commands.py]: mm.. please place these imports at the top of the file

**Review by nemesifier — CHANGES_REQUESTED** [review]
ops, I hadn't spotted those imports at first

File: django_ipam/tests/base/test_commands.py (hunk 2 above)
Comment (cking100) [reply django_ipam/tests/base/test_commands.py]: Is it a bad practice to do so? I have seen people doing it . Will try avoiding it from next time if its not a good practice

Comment (nemesifier) [reply django_ipam/tests/base/test_commands.py]: it's bad unless you have a very specific need for doing it, which usually boils down to avoiding circular imports, and even in these cases, it should be used as a last resort

**Review by nemesifier — APPROVED** [review]

**Pull request merged**